packages/dmn-editor/src/dataTypes/ImportJavaClasses.tsx

```diff
@@ -0,0 +1,179 @@
+  JavaCodeCompletionService,
+  useImportJavaClassesWizardI18n,
+  useLanguageServerAvailable,
+} from "@kie-tools/import-java-classes-component";
+import { addTopLevelItemDefinition as _addTopLevelItemDefinition } from "../mutations/addTopLevelItemDefinition";
+import { JavaClassConflictOptions, useImportJavaClasses } from "./useImportJavaClasses";
+import { Radio } from "@patternfly/react-core/dist/esm/components/Radio";
+
+const ImportJavaClassesWrapper = ({
+  javaCodeCompletionService,
+}: {
+  javaCodeCompletionService: JavaCodeCompletionService;
+}) => {
+  const { handleConflictAction, handleImportJavaClasses, conflictsClasses, isConflictsOccured } =
+    useImportJavaClasses();
+  return (
+    <>
+      <ImportJavaClasses
+        loadJavaClassesInDataTypeEditor={handleImportJavaClasses}
+        javaCodeCompletionService={javaCodeCompletionService}
+      />
+      {isConflictsOccured && conflictsClasses?.length > 0 && (
+        <ImportJavaClassNameConflictsModal
+          isOpen={isConflictsOccured}
+          handleConfirm={handleConflictAction}
+          conflictsNames={conflictsClasses}
+        />
+      )}
+    </>
+  );
+};
+
+const ImportJavaClassesDropdownItem = ({
+  javaCodeCompletionService,
+  ...props
+}: React.ComponentProps<typeof DropdownItem> & { javaCodeCompletionService: JavaCodeCompletionService }) => {
+  const { i18n } = useImportJavaClassesWizardI18n();
+  const { isLanguageServerLoading, isLanguageServerDisabled, isLanguageServerError } =
+    useLanguageServerAvailable(javaCodeCompletionService);
+  const defineTooltipMessage = React.useCallback(() => {
+    if (isLanguageServerDisabled) {
+      return i18n.modalButton.disabledMessage;
+    } else if (isLanguageServerError) {
+      return i18n.modalButton.errorMessage;
+    }
+    return undefined;
+  }, [
+    isLanguageServerDisabled,
+    isLanguageServerError,
+    i18n.modalButton.disabledMessage,
+    i18n.modalButton.errorMessage,
+  ]);
+  return (
+    <>
+      {defineTooltipMessage() ? (
+        <Tooltip content={defineTooltipMessage()}>
+          <DropdownItem
+            style={{ minWidth: "240px" }}
+            icon={isLanguageServerLoading ? <Spinner size="md" /> : <PlusIcon />}
+            isDisabled={isLanguageServerDisabled || isLanguageServerLoading}
+            {...props}
+          >
+            {i18n.modalButton.text}
+          </DropdownItem>
+        </Tooltip>
+      ) : (
+        <DropdownItem
+          style={{ minWidth: "240px" }}
+          icon={isLanguageServerLoading ? <Spinner size="md" /> : <PlusIcon />}
+          isDisabled={isLanguageServerDisabled || isLanguageServerLoading}
+          {...props}
+        >
+          {i18n.modalButton.text}
+        </DropdownItem>
+      )}
+    </>
+  );
+};
+
+const ImportJavaClassNameConflictsModal = ({
+  isOpen,
+  handleConfirm,
+  conflictsNames,
+}: {
+  isOpen: boolean;
+  handleConfirm: (options: JavaClassConflictOptions) => void;
+  conflictsNames: JavaClass[];
```

Comment:
```suggestion
  confliectedJavaClasses: JavaClass[];
```

I think this better represents what the prop actually is, WDYT?

Reply:
`confliectedJavaClasses` will be a good one. Thanks. I’ve renamed the variable.